sonar-python-plugin/src/main/java/org/sonar/plugins/python/indexer/PythonIndexer.java

```diff
@@ -89,6 +89,10 @@ public boolean canBeScannedWithoutParsing(InputFile inputFile) {
     return false;
   }
 
+  public boolean canBeFullyScannedWithoutParsing(InputFile inputFile) {
+    return false;
+  }
```

Comment:
As discussed: having both a `canBeScannedWithoutParsing` and `canBeFullyScannedWithoutParsing` is a bit confusing for me. Maybe we could rename `canBeFullyScannerWithoutParsing` to something like `dependsOnChangedFile` and the flip the usage?

Reply:
As discussed, I kind of like the fact that the notion of `partiallySkippable` and `fullySkippable` introduces a bit of terminology (even if it's admittedly a bit of a goofy one) that helps make communication a bit clearer.

I renamed the `canBeScannedWithoutParsing` to `canBePartiallyScanWithoutParsing` and added comments explaining the meaning of those concepts so that it's hopefully less confusing.